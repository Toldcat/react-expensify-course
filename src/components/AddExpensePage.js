import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startAddExpense } from '../actions/expenses'

export const AddExpensePage = props => (
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      onSubmit={expense => {
        props.onSubmit(expense)
        //redirect after expenses have been added
        props.history.push('/')
      }}
    />
  </div>
)

const mapDispatchToProps = dispatch => ({
  onSubmit: expense => dispatch(startAddExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage)
