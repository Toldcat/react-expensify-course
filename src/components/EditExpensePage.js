import React from 'react'
import { connect } from 'react-redux'
import { editExpense, removeExpense } from '../actions/expenses'
import ExpenseForm from './ExpenseForm'

export const EditExpensePage = props => (
  <div>
    <ExpenseForm
      expense={props.expense}
      onSubmit={expense => {
        //Dispatch edit expense action
        props.dispatch(editExpense(props.expense.id, expense))
        //Redirect to dashboard
        props.history.push('/')
      }}
    />
    <button
      onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id }))
        //Redirect to dashboard
        props.history.push('/')
      }}
    >
      Remove
    </button>
  </div>
)

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      expense => expense.id === props.match.params.id
    )
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: data => dispatch(removeExpense(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
