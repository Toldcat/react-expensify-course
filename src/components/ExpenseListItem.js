import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

export const ExpenseListItem = ({ description, amount, createdAt, id }) => (
  <Link className='list__item' to={`/edit/${id}`}>
    <div>
      <h3 className='list__item-title'>{description}</h3>
      <span className='list__item-subtitle'>
        {moment(createdAt).format('Do [of] MMMM, YYYY')}
      </span>
    </div>
    <h3 className='list__item-data'>
      {'Â£' + numeral(amount / 100).format('0,0.00')}
    </h3>
  </Link>
)

export default ExpenseListItem
