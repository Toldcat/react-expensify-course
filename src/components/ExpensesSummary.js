import React from 'react'
import { connect } from 'react-redux'
import expensesTotal from '../selectors/expenses-total'
import visibleExpenses from '../selectors/expenses'

export const ExpensesSummary = props => (
  <div>
    <p>
      Showing {props.expenses.length}{' '}
      {props.expenses.length === 1 ? 'expense' : 'expenses'} totalling Â£
      {props.count / 100}
    </p>
  </div>
)

const mapStateToProps = state => {
  return {
    expenses: visibleExpenses(state.expenses, state.filters),
    count: expensesTotal(visibleExpenses(state.expenses, state.filters))
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
