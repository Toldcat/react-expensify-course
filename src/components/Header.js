import React from 'react'
import { NavLink } from 'react-router-dom'
import { startLogout } from '../actions/auth'
import { connect } from 'react-redux'

export const Header = props => (
  <header>
    <h1>Expensify</h1>

    <NavLink exact to='/dashboard' activeClassName='selected'>
      Dashboard
    </NavLink>
    <NavLink to='/create' activeClassName='selected'>
      Create New Expense
    </NavLink>
    <button onClick={props.startLogout}>Logout</button>
  </header>
)

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)
