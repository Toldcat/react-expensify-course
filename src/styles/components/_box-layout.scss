.box-layout {
  background: url('/images/bg.jpg');
  background-size: cover;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;

  &__box {
    background-color: rgba(225, 225, 225, 0.85);
    border-radius: 3px;
    text-align: center;
    width: 30rem;
    padding: $l-size $m-size;
  }

  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }
}
