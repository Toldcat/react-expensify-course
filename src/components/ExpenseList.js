import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import visibleExpenses from '../selectors/expenses'

export const ExpenseList = props => (
  <div className='content-container'>
    <div className='list'>
      <div className='show-on-mobile'>Expenses</div>
      <div className='show-on-desktop'>Expense</div>
      <div className='show-on-desktop'>Amount</div>
    </div>
    <div className='list__body'>
      {props.expenses.length === 0 ? (
        <div className='list__item list__item-message'>
          <span>No Expenses Found</span>
        </div>
      ) : (
        props.expenses.map(expense => (
          <ExpenseListItem key={expense.id} {...expense} />
        ))
      )}
    </div>
  </div>
)

const mapStateToProps = state => {
  return {
    expenses: visibleExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(ExpenseList)
