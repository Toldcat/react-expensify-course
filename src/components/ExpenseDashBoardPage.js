import React from 'react'
import ExpenseList from './ExpenseList'
import ExpensesSummary from './ExpensesSummary'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashBoardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList />
  </div>
)

export default ExpenseDashBoardPage
