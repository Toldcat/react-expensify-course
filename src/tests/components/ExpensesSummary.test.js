import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expensesTestData from '../fixtures/expenses'

test('should render Expenses Summary component', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expensesTestData} />)
  expect(wrapper).toMatchSnapshot()
})
