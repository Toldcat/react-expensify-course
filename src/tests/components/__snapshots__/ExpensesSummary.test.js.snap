// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary component 1`] = `
<div>
  <p>
    Showing 
    3
     
    expenses
     totalling Â£
    NaN
  </p>
</div>
`;
